src/azul/collections.py

```diff
@@ -271,3 +273,64 @@ def to_dict(self):
             k: v.to_dict() if isinstance(v, NestedDict) else v
             for k, v in self.items()
         }
+
+
+class OrderedSet(MutableSet[K]):
+    """
+    A mutable set that maintains insertion order. It is not a sequence.
+
+    >>> OrderedSet() == set()
+    True
+    >>> 'a' in OrderedSet(['a', 'b'])
+    True
+    >>> OrderedSet(['a', 'b', 'c', 'b'])
+    OrderedSet(['a', 'b', 'c'])
+    """
+
+    def __init__(self, members: Iterable[K] = (), /) -> None:
+        self.inner: dict = dict.fromkeys(members)
```

Comment:
Another opportunity to improve the type annotations:

```suggestion
        self.inner: dict[K, None] = dict.fromkeys(members)
```